sources/packages/backend/apps/queue-consumers/src/processors/schedulers/sfas-integration/sfas-integration.scheduler.ts

```diff
@@ -0,0 +1,43 @@
+import { InjectQueue, Process, Processor } from "@nestjs/bull";
+import { Job, Queue } from "bull";
+import { SFASProcessingResultQueueOutDTO } from "./models/sfas-integration.dto";
+import { BaseScheduler } from "../base-scheduler";
+import { QueueNames } from "@sims/utilities";
+import { QueueService } from "@sims/services/queue";
+import { SFASIntegrationProcessingService } from "@sims/integrations/sfas-integration";
+
+/**
+ * Process all SFAS integration files from the SFTP location.
+ */
+@Processor(QueueNames.SFASIntegration)
+export class SFASIntegrationScheduler extends BaseScheduler<void> {
+  constructor(
+    @InjectQueue(QueueNames.SFASIntegration)
+    schedulerQueue: Queue<void>,
+    queueService: QueueService,
+    private readonly sfasIntegrationProcessingService: SFASIntegrationProcessingService,
+  ) {
+    super(schedulerQueue, queueService);
+  }
+
+  /**
+   * Process all SFAS integration files from the SFTP
+   * and update the database with processed records.
+   * @param job SFAS integration job.
+   * @returns processing result.
+   */
+  @Process()
+  async processSFASIntegrationFiles(
+    job: Job<void>,
+  ): Promise<SFASProcessingResultQueueOutDTO[]> {
+    await job.log("Processing SFAS integration files...");
```

Comment:
QueueProcessSummary?

Reply:
As per discussion, QueueProcessSummary cannot be directly replaced here. I removed the QueueOutDTO from interface name.